### DISCUSSION TITLE
Query related to packagerevision CR

### DISCUSSION THREAD

**Discussion by SamarSidharth** [kptdev/kpt#4069] [General]
There are few CR with GV as `config.porch.kpt.dev/v1alpha1` and few with `porch.kpt.dev/v1alpha1`.
The one with `porch.kpt.dev/v1alpha1` are not visble in the `kubectl get crd` output

So, I am little confused because I cannot connect the CR with CRD for resources like packagerevisionresources, packagerevisions.

> kubectl api-resources| grep porch
functions                                      config.porch.kpt.dev/v1alpha1          true         Function
packagerevs                                    config.porch.kpt.dev/v1alpha1          true         PackageRev
packagevariants                                config.porch.kpt.dev/v1alpha1          true         PackageVariant
packagevariantsets                             config.porch.kpt.dev/v1alpha2          true         PackageVariantSet
remoterootsyncsets                             config.porch.kpt.dev/v1alpha1          true         RemoteRootSyncSet
repositories                                   config.porch.kpt.dev/v1alpha1          true         Repository
rootsyncsets                                   config.porch.kpt.dev/v1alpha1          true         RootSyncSet
workloadidentitybindings                       config.porch.kpt.dev/v1alpha1          true         WorkloadIdentityBinding
functions                                      porch.kpt.dev/v1alpha1                 true         Function
packagerevisionresources                       porch.kpt.dev/v1alpha1                 true         PackageRevisionResources
packagerevisions                               porch.kpt.dev/v1alpha1                 true         PackageRevision
packages                                       porch.kpt.dev/v1alpha1                 true         Package

> kubectl get crd -o wide
NAME                                            CREATED AT
configmanagements.configmanagement.gke.io       2023-10-13T03:43:48Z
functions.config.porch.kpt.dev                  2023-10-13T03:43:46Z
packagerevs.config.porch.kpt.dev                2023-10-13T03:43:46Z
packagevariants.config.porch.kpt.dev            2023-10-13T03:43:46Z
packagevariantsets.config.porch.kpt.dev         2023-10-13T03:43:46Z
remoterootsyncsets.config.porch.kpt.dev         2023-10-13T03:43:47Z
repositories.config.porch.kpt.dev               2023-10-13T03:43:47Z
rootsyncsets.config.porch.kpt.dev               2023-10-13T03:43:47Z
workloadidentitybindings.config.porch.kpt.dev   2023-10-13T03:43:47Z

**Comment by johnbelamaric** [comment]
Yeah, we need to rationalize the groups we are using.

Right now, PackageRevisions are split across a CR and a separate resource served up by the aggregated API server. I believe @mortent has a plan to improve this situation. Is there a doc for that Morten?

**Comment by natasha41575** [comment]
There is an open issue for moving packagerevisions to a CRD: https://github.com/kptdev/kpt/issues/3630, that enumerates the technical details of what would be required here. packagerevisionresources will still likely need to be an aggregated API IIRC.

**Comment by SamarSidharth** [comment]
Thanks @natasha41575 and @johnbelamaric for the clarification. In this case the schema validation of the resources belonging to GV `porch.kpt.dev/v1alpha1` would be handled by the porch-server pod ? The reason I am asking this is because sometime back I face an issue where I wasn't able to list the packagerevisions and packagerevisionresources resources.

Also let me know if I can assist on some smaller tasks for #3630, I will be happy to help.

**Reply by johnbelamaric** [reply]
Thanks, that could be really helpful. I think we still have to figure out exactly what we want. Let's continue the discussion on #3630
